### PR TITLE
Refactor/object position dimensions

### LINKED ISSUE #90: Add Model Loading to Server + Encapsulate Object Position
**Task Description:**

- [ ] Update `ServerGameState` to read object models and populate a list of `Model`s (with the model size information).
- [ ] Add a `Model` struct to the server that maintains model dimension info
- [ ] Add a `ModelType` enum that maps each model to a particular `Model` (?)
- [ ] Add a `ModelType` or `Model` field to `Object` and `SharedObject`
    - [ ] Encapsulate Physics position + collider position + model dimension into single setters and getters in the `Object` class (i.e., to update object position, must set it via getters / setters to avoid forgetting to set a part of the object position correctly)

### PR DESCRIPTION
**Pull request by gilkeidar**
Refactored `Object` position + dimensions information to avoid state duplication which required careful syncing of different fields.

Here is the new position + dimensions `Object` paradigm:

1. `Object` now has a `ModelType` enum field that specifies the rendering model that this `Object` should render as by the client. (For the DM, this may require setting `ModelType::None` (which doesn't exist currently) or the client can simply hard-code the behavior that the DM shouldn't be rendered regardless of its specified model type)
2. The `Collider` field in the `Physics` struct has been replaced: instead of a `Collider *` to one of the derived `Collider` classes (`BoxCollider` or `SphereCollider`),  it now has a `Collider` enum field which can be either `Box` or `Sphere`, with collision detection placed in global functions declared in `server/game/collider.hpp`. This means colliders have no state and utilize the `SharedPhysics` fields of `corner` and `dimensions` to detect collisions.
3. The `position` field in `SharedPhysics` has been removed.
4. The `dimensions` field in `SharedSolidSurface` has been removed.

The `Object` class now has a new static field: `static std::unordered_map<ModelType, glm::vec3> models` which maps the `ModelType` enum values to their model's default dimensions. Currently, these are all hard-coded in `object.cpp`, but this can be replaced by the server reading the model object files upon initialization and populating this hash map dynamically.

Note: The hard-coded dimensions for `ModelType::Player` and `ModelType::WarrenBear` do **NOT** currently correspond to the actual model dimensions! Their values are my current guesses as I'm not sure how to extract the models' dimension data. If the player / warren bear do not render properly / in the right positions, this is likely why.

Closes #90

### CODE REVIEW

File: src/client/client.cpp [hunk 1]
```diff
@@ -286,13 +286,18 @@ void Client::draw() {
...
+                    glm::vec3 pos = sharedObject->physics.corner + (sharedObject->physics.dimensions / 2.0f);
                     pos.y += PLAYER_EYE_LEVEL;
                     cam->updatePos(pos);
                     break;
                 }
                 auto lightPos = glm::vec3(0.0f, 10.0f, 0.0f);
-                auto player_pos = glm::vec3(sharedObject->physics.position.x, sharedObject->physics.position.y + 0.1, sharedObject->physics.position.z);
+                //auto player_pos = glm::vec3(sharedObject->physics.position.x, sharedObject->physics.position.y + 0.1, sharedObject->physics.position.z);
+
+                auto player_pos = sharedObject->physics.corner + (sharedObject->physics.dimensions / 2.0f);
```
Comment (Tyler-Lentz) [inline src/client/client.cpp]: Maybe we can make a function on the Physics struct called getCenterPos that does this calculation for us?

**Review by Tyler-Lentz — COMMENTED** [review]
I'm doing this on my phone while riding the bus, so I'll be able to test on my computer in a bit. I just left the one potential tiny comment which might be nice to have. Also, if we could remove the commented out code that would be nice, so we don't get confused later. I'll drop the approve in a bit once I verify it works on my laptop (which it should but just want to be sure)

Overall this is incredibly helpful! Thanks so much for grinding this out!

**Review by jhparkt — APPROVED** [review]

**Review by Tyler-Lentz — APPROVED** [review]
just pushed a fix for the client side UI which had broken in a previous PR and we didn't notice. this should be good to merge once we clear out the commented code

yipee-1

File: src/client/client.cpp (hunk 1 above)
Comment (gilkeidar) [reply src/client/client.cpp]: Added!

**Review by atar13 — APPROVED** [review]
Looks great. I'm liking this new structure.

**Pull request merged**